lib/PreviewProvider.js

```diff
@@ -173,40 +181,74 @@ PreviewProvider.prototype = {
       return links;
     }
 
-    return links.map(link => {
+    let now = Date.now();
+
+    let results = links.map(link => {
       if (link && link.cacheKey && ss.storage.embedlyData[link.cacheKey]) {
+        ss.storage.embedlyData[link.cacheKey].accessTime = now;
         return Object.assign({}, ss.storage.embedlyData[link.cacheKey], link);
       } else {
         return null;
       }
     })
     .filter(link => link);
+
+    // gives the opportunity to have at least one run with an old
+    // preview cache. This is for the case where one hasn't opened
+    // the browser since `cacheTTL`
+    this.cleanUpCacheMaybe();
```

Comment:
I assume this should never take too much time? It is blocking the returning of cached results. Any reason it wasn't done as part of the periodic updates instead?